PseudoInst: Split huge switch into functions

agnat opened the pull request:
Sorry 'bout the mess.

dylanmckay commented:
I was just working on this file yesterday and was thinking how much of a mess it was.

I especially like how you have the `expand` function specialised using templates for the different opcodes.

If you can squash the PR, I'll merge :)

dylanmckay commented:
I fixed a bug in `AVRExpandPseudoInsts.cpp` in 9d70a5c2d428a1a7149fa4484b96fdfbb46a0633 which caused a merge conflict.

agnat commented:
Squashed and rebased on avr-support. CI is happy too...

> I especially like how you have the expand function specialised...

Thanks. Glad you like it. ;)
